sdk/resourcemanager/Azure.ResourceManager/docs/MigrationGuide.md

```diff
@@ -599,6 +599,61 @@ VirtualMachineResource virtualMachine = virtualMachineOperation.Value;
 
 Finally, as it can be seen here, from the resource group you can get the Virtual Machine collection and create a new one using the `VirtualMachineData` for the parameters.
 
+### Update a Virtual Machine
+
+**Create a new object of VirtualMachineData**
+```C#
+var virtualMachineDataToModify = new VirtualMachineData(AzureLocation.EastUS)
+{
+    NetworkProfile = new VirtualMachineNetworkProfile()
+    {
+        NetworkInterfaces =
+        {
+            new VirtualMachineNetworkInterfaceReference()
+            {
+                Id = networkInterface.Id,
+                Primary = false,
+            }
+        }
+    }
+};
+var virtualMachineModify = (await virtualMachines.CreateOrUpdateAsync(WaitUntil.Completed, virtualMachineName, virtualMachineDataToModify)).Value;
```

Comment:
does this really work?
The PUT method is actually overriding the instance on Azure, therefore it usually requires us to send the full data. In the case of virtual machine, if we want to change the network interfaces of a vm using a PUT method, we should:
1. Call Get to get the existing VM data.
2. modify the VM data, in your case, change the `NetworkInterfaces`, since this is a list, we might need to call `Clear` first and then use `Add` to add new entries if your purpose is to completely change the list.
3. Call PUT (CreateOrUpdate) using your modified VM data.

Reply:
Your current way is the "PATCH" way to do an update.